test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -3984,7 +3987,34 @@ TEST_P(ExtProcIntegrationTest, MappedAttributeBuilder) {
                           return true;
                         });
 
-  handleUpstreamRequest();
+  handleUpstreamRequestWithTrailer();
+
+  // Handle response headers message.
+  processGenericMessage(*grpc_upstreams_[0], false,
+                        [](const ProcessingRequest& req, ProcessingResponse& resp) {
+                          // Add something to the response so the message isn't seen as spurious
+                          resp.mutable_response_headers();
+
+                          EXPECT_TRUE(req.has_response_headers());
+                          EXPECT_EQ(req.attributes().size(), 1);
```

Comment:
With one req.attributes and one resp.attributes,  so I guess the for loop above is not tested.